operators/src/pro/ml/xgboost.rs

```diff
@@ -0,0 +1,628 @@
+            vec_of_rqps,
+            self.model_file_path.clone(),
+            self.no_data_value,
+        ))))
+    }
+}
+
+struct XgboostProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+{
+    sources: Vec<Q>,
+    model_file: PathBuf,
+    no_data_value: Option<f32>,
+}
+
+impl<Q, P> XgboostProcessor<Q, P>
+where
+    Q: RasterQueryProcessor<RasterType = P>,
+    P: Pixel,
+{
+    pub fn new(sources: Vec<Q>, model_file_path: PathBuf, no_data_value: Option<f32>) -> Self {
+        Self {
+            sources,
+            model_file: model_file_path,
+            no_data_value,
+        }
+    }
+
+    async fn process_tile_async(
+        &self,
+        mut bands_of_tile: Vec<Result<RasterTile2D<P>, crate::error::Error>>,
+        pool: Arc<ThreadPool>,
+    ) -> Result<RasterTile2D<PixelOut>> {
+        let tile = bands_of_tile.pop().ok_or(error::Error::EmptyInput)??;
+
+        // gather the data
+        let grid_shape = tile.grid_shape();
+        let n_bands = bands_of_tile.len() as i32 + 1; // +1 because of .pop()
+        let props = tile.properties.clone(); // = &tile.properties;
+        let time = tile.time;
+        let tile_position = tile.tile_position;
+        let global_geo_transform = tile.global_geo_transform;
+
+        let model = self.model_file.clone();
+        let ndv = self.no_data_value.or(Some(f32::NAN));
+
+        let predicted_grid: geoengine_datatypes::raster::Grid<GridShape<[usize; 2]>, f32> =
+            crate::util::spawn_blocking(move || {
+                // put the tile back into place
+                bands_of_tile.insert(0, Ok(tile));
+                // extract the actual data from the tiles
+                let pixels: Vec<_> = bands_of_tile
+                    .into_iter()
+                    .flat_map(|band| {
+                        let band_ok = match band {
+                            Ok(x) => x,
+                            Err(err) => {
+                                panic!("Failed to get BaseTile with error msg: \n{:?}", err)
+                            }
+                        };
+
+                        let mat_tile = band_ok.into_materialized_tile();
+
+                        mat_tile
+                            .grid_array
+                            .inner_grid
+                            .data
+                            .into_iter()
+                            .map(num_traits::AsPrimitive::as_)
+                            .collect::<Vec<f32>>()
+                    })
+                    .collect();
+
+                // let pixels: Vec<_> = rasters.into_iter().collect();
+                let model_content = std::fs::read_to_string(model).unwrap();
+
+                process_tile(
+                    &pixels,
+                    &pool,
+                    model_content.as_bytes(),
+                    grid_shape,
+                    n_bands as usize,
+                    ndv.unwrap(),
```

Comment:
also here: don't unwrap but return an Error. Maybe you can also check this during initialization and unpack the option there so that you know here that there is a value.

Reply:
I removed the option from that with https://github.com/geo-engine/geoengine/pull/639/commits/503b946b55d52e2613f672271a28f1bb79ca5f17